### PR TITLE
Post regrid actions for physics on restart with regrid

### PR DESCRIPTION
**Pull request by marchdf**
## Summary

When restarting a simulation with more levels than the checkpoint, regrid is called. This is typically not a problem, except if the fillpatch from coarse is not good enough and the physics class does some non-trivial post-regrid action. This was the case for the terrain fields. I kept it simple by only doing the physics post-regrid to avoid any other unintended consequence.

I accidentally broke the terrain fields on a restart with more levels in #1353 when I moved their initialization from post_init to initialize_fields. This was the right move but didn't account for restarting with more levels. Additionally, having the reinit in the post-regrid means that this will now work with dynamic mesh refinement. 

## Pull request type

Please check the type of change introduced:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Checklist

The following is included:

- [ ] new unit-test(s)
- [x] new regression test(s)
- [ ] documentation for new capability

This PR was tested by running:

- the unit tests
  - [ ] on GPU 
  - [x] on CPU 
- the regression tests
  - [ ] on GPU 
  - [x] on CPU

### CODE REVIEW

**Comment by hgopalan** [comment]
Thank you for the quick fix.

**Comment by hgopalan** [comment]
Works perfectly and no issues with convergence for RANS. 
AMR_RANS

**Comment by marchdf** [comment]
Thanks for checking, @hgopalan !

**Review by hgopalan — APPROVED** [review]

File: amr-wind/physics/TerrainDrag.H [hunk 1]
```diff
@@ -28,14 +28,12 @@ public:
 
     void post_init_actions() override {}
 
-    void post_regrid_actions() override {}
+    void post_regrid_actions() override;
```
Comment (moprak-nrel) [inline amr-wind/physics/TerrainDrag.H:31]: Any reason why this is changed back to `;` vs `{}`

Comment (marchdf) [reply amr-wind/physics/TerrainDrag.H:31]: because I actually define this function now with something non-trivial in the cpp file.

**Review by moprak-nrel — APPROVED** [review]

**Pull request merged**